packages/dev/inspector-v2/src/components/properties/particles/particleSystemProperties.tsx

```diff
@@ -1,109 +1,1252 @@
+    );
+
+    const loadFromSnippetServer = useCallback(() => {
+        if (!scene) {
+            alert("No scene available.");
+            return;
+        }
+
+        // Prompt for a snippet id (minimal UX).
+
+        const requestedSnippetId = window.prompt("Please enter the snippet ID to use");
+        const trimmed = requestedSnippetId?.trim();
+        if (!trimmed) {
+            return;
+        }
+
+        const request = new XMLHttpRequest();
+        request.onreadystatechange = () => {
+            if (request.readyState !== 4) {
+                return;
+            }
+
+            if (request.status !== 200) {
+                alert("Unable to load your particle system");
+                return;
+            }
+
+            try {
+                const responseObject = ParseJsonLoadContents(request.responseText);
+                if (!responseObject) {
+                    alert("Unable to load your particle system");
+                    return;
+                }
+
+                applyParticleSystemJsonToSystem(responseObject);
+                system.snippetId = trimmed;
+            } catch (e) {
+                alert("Unable to load your particle system: " + e);
+            }
+        };
+
+        request.open("GET", ParticleHelper.SnippetUrl + "/" + trimmed.replace(/#/g, "/"), true);
+        request.send();
+    }, [applyParticleSystemJsonToSystem, scene, system]);
+
+    const saveToSnippetServer = useCallback(() => {
+        // Serialize once and post as snippet payload.
+        const content = JSON.stringify(system.serialize(true));
+
+        const xmlHttp = new XMLHttpRequest();
+        xmlHttp.onreadystatechange = () => {
+            if (xmlHttp.readyState !== 4) {
+                return;
+            }
+
+            if (xmlHttp.status !== 200) {
+                alert("Unable to save your particle system");
+                return;
+            }
+
+            try {
+                const snippet = JSON.parse(xmlHttp.responseText);
+                const oldId = system.snippetId || "_BLANK";
+                system.snippetId = snippet.id;
+                if (snippet.version && snippet.version !== "0") {
+                    system.snippetId += "#" + snippet.version;
+                }
+
+                // Copy to clipboard when available.
+                if (navigator.clipboard) {
+                    void navigator.clipboard.writeText(system.snippetId);
+                }
+
+                const windowAsAny = window as any;
+                if (windowAsAny.Playground && oldId) {
+                    // Update Playground code when snippet id is hard-coded.
+                    windowAsAny.Playground.onRequestCodeChangeObservable.notifyObservers({
+                        regex: new RegExp(`ParticleHelper.ParseFromSnippetAsync\\("${oldId}`, "g"),
+                        replace: `ParticleHelper.ParseFromSnippetAsync("${system.snippetId}`,
+                    });
+                }

```

Comment:
I know we have this pattern a handful of times in Inspector v1, but I think for v2 we don't want to propagate this pattern. Instead, we should probably have a service that fires this event, and then Playground can add a service that consumes this service, registers for the event, and does its own code patching. We could try to do that with this PR, or we could just add a comment saying we should do this and don't propagate this pattern. Let me have a look and see how much work it would be to do this.

Reply:
@ryantrem should we merge and addreess later or is it better to do the change now ?